airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -228,13 +241,45 @@ private static AirbyteStream removeIncrementalWithoutPk(final AirbyteStream stre
   }
 
   @Override
-  public Set<JdbcPrivilegeDto> getPrivilegesTableForCurrentUser(final JdbcDatabase database, final String schema) throws SQLException {
+  public Set<JdbcPrivilegeDto> getPrivilegesTableForCurrentUser(final JdbcDatabase database,
+                                                                final String schema)
+      throws SQLException {
     return database.query(connection -> {
       final PreparedStatement ps = connection.prepareStatement(
-          "SELECT DISTINCT table_catalog, table_schema, table_name, privilege_type\n"
-              + "FROM   information_schema.table_privileges\n"
-              + "WHERE  grantee = ? AND privilege_type = 'SELECT'");
-      ps.setString(1, database.getDatabaseConfig().get("username").asText());
+          // c.relkind = 'm' means Materialized View
+          // c.relacl is null or s[2] = 'r' means either null (all grants) or "r" - grants for read
+          // coalesce ('SELECT') as privilege_type\n" is set as it to match Union query, verification is done
+          // is "where" clause
```

Comment:
This is really helpful.

- Can you move the comments to the SQL code directly? Like

  ```sql
  -- 'm' mean Materialized View
  AND    c.relkind = 'm'
  AND    (
                 -- all grants
                 c.relacl IS NULL
                 -- read grant
          OR     s[2] = 'r');
  ```

  In this way it is easier to tell which statement a comment is about.

- The comment about `coalesce ('SELECT') as privilege_type` is not clear. Can you rephrase it?

Reply:
Updated, thanks

Reply:
/test connector=connectors/source-postgres-strict-encrypt